modelmapper/signature.py

```diff
@@ -0,0 +1,72 @@
+from collections import Mapping
+from decimal import Decimal
+
+import mmh3
+
+
+def generate_row_signature(row, model=None, ignore_fields=None):
+    """ Generates a 64 bit hash of the given row
+        Arguments:
+            row: A dictionary or list of tuples
+            model: Sqlalchemy model that coresponses to the given row
+            ignore_fields: Fields to be ignored in the calcuation of the hash
+        Returns:
+            Integer: the hash value of the given row
+    """
+    if isinstance(row, list):
+        row_dict = dict(row)
```

Comment:
Are we converting a list to a dictionary? What do we get?

Reply:
This was in case of a list of tuples being provided. I can add a type check the elements in the list to make it more robust